app/api/logging/audit.py

```diff
@@ -0,0 +1,71 @@
+#
+# Application-level audit logging.
+#
+# See https://docs.python.org/3/library/audit_events.html
+# https://docs.python.org/3/library/sys.html#sys.addaudithook
+# https://www.python.org/dev/peps/pep-0578/
+#
+
+import logging
+import sys
+from typing import Any
+
+import api.logging
+import api.util.collections.dict
+
+logger = api.logging.get_logger(__name__)
+
+AUDIT = 32
+
+logging.addLevelName(AUDIT, "AUDIT")
+
+
+def init_security_logging() -> None:
+    sys.addaudithook(audit_hook)
+
+
+IGNORE_AUDIT_EVENTS = {
+    "builtins.id",
+    "code.__new__",
+    "compile",
+    "exec",
+    "import",
+    "marshal.loads",
+    "object.__getattr__",
+    "object.__setattr__",
+    "os.listdir",
+    "os.scandir",
+    "socket.__new__",
+    "sys._current_frames",
+    "sys._getframe",
+    "sys.settrace",  # interferes with PyCharm debugger
+}
+
+
+def audit_hook(event_name: str, args: tuple[Any, ...]) -> None:
+    if event_name in IGNORE_AUDIT_EVENTS:
+        return
+    if event_name == "open" and isinstance(args[0], str) and "/__pycache__/" in args[0]:
+        return
+    if event_name == "os.chmod" and type(args[0]) is int:
+        # Gunicorn generates a high volume of these events in normal operation (see workertmp.py)
+        return
+    # Disable audit log for timezone files.
```

Comment:
nit: I generally prefer to put comments inside the if block indent!

```
if event_name == "open" and isinstance(args[0], str) and "/pytz/" in args[0]:
    # Disable audit log for timezone files.
    return
```

Reply:
also, I'd appreciate an explanation in the comment as to _why_ we're ignoring timezone files!